server/src/main/java/io/crate/planner/node/ddl/CreateRolePlan.java

```diff
@@ -53,19 +69,77 @@ public StatementType type() {
     public void executeOrFail(DependencyCarrier dependencies,
                               PlannerContext plannerContext,
                               RowConsumer consumer,
-                              Row params, SubQueryResults subQueryResults) throws Exception {
-        SecureHash newPassword = UserActions.generateSecureHash(
+                              Row params,
+                              SubQueryResults subQueryResults) throws Exception {
+
+        Map<String, Object> properties = validate(
             createRole.properties(),
-            params,
             plannerContext.transactionContext(),
-            plannerContext.nodeContext());
+            plannerContext.nodeContext(),
+            params,
+            subQueryResults
+        );
+        SecureHash newPassword = UserActions.generateSecureHash(properties);
 
         if (createRole.isUser() == false && newPassword != null) {
             throw new UnsupportedOperationException("Creating a ROLE with a password is not allowed, " +
                                                     "use CREATE USER instead");
         }
 
-        roleManager.createRole(createRole.roleName(), createRole.isUser(), newPassword)
+        // Map is always not null to simplify streaming.
+        // It's safe to create a JwtProperties object as user input already passed validation.
+        Map<String, Object> jwtPropertiesMap = Maps.getOrDefault(properties, JWT_PROPERTY_KEY, Map.of());
+        JwtProperties jwtProperties = null;
+        if (jwtPropertiesMap.isEmpty() == false) {
+            String iss = Maps.get(jwtPropertiesMap, "iss");
+            String username = Maps.get(jwtPropertiesMap, "username");
+            jwtProperties = new JwtProperties(iss, username);
+        }
+
+        roleManager.createRole(createRole.roleName(), createRole.isUser(), newPassword, jwtProperties)
             .whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
     }
+
+    public static Map<String, Object> validate(GenericProperties<Symbol> genericProperties,
+                                               TransactionContext txnCtx,
+                                               NodeContext nodeContext,
+                                               Row params,
+                                               SubQueryResults subQueryResults) {
+        Function<? super Symbol, Object> eval = x -> SymbolEvaluator.evaluate(
+            txnCtx,
+            nodeContext,
+            x,
+            params,
+            subQueryResults
+        );
+        Map<String, Object> properties = genericProperties.map(eval).properties();
+        for (var entry : properties.entrySet()) {
+            var key = entry.getKey();
+            switch (key) {
+                case JWT_PROPERTY_KEY:
```

Comment:
Couldn't do this `.get()` lookups instead of iterating over all entries? Seems a bit odd to iterate and then check for specific keys.
Detecting if there are invalid properties could be done via a validSettings.containsAll or something like that.

Reply:
Yep, currently JWT is looked up twice. Fixed it in https://github.com/crate/crate/pull/15515/commits/7a33d1c6bbd39aef81b25c909d9d89813ce93ee8

Iterating over top-level keys for validation is required but now it's simpler as I moved JWT specific things to JwtProperties - related to comment below 
> JwtProperties.parse or something like that?